src/transport/SessionManager.cpp

```diff
@@ -573,20 +579,162 @@ void SessionManager::OnMessageReceived(const PeerAddress & peerAddress, System::
+    Transport::PeerAddress peerAddr = conn->mPeerAddr;
+    peerAddr.ToString(peerAddrBuf);
+
+    if (appConnCbCtxt->connReceivedCb != nullptr)
+    {
+        appConnCbCtxt->connReceivedCb(conn);
+    }
+    else
+    {
+        ChipLogProgress(Inet, "%s: Received connection request from %s, but no registered handler. Disconnecting.", peerAddrBuf,
+                        __func__);
+
+        // Close the connection
+        Disconnect(conn);
+    }
+}
+
+void SessionManager::HandleConnectionComplete(Transport::ActiveConnectionState * conn, CHIP_ERROR conErr)
+{
+    Transport::AppConnCallbackCtxt * appConnCbCtxt = nullptr;
+    char peerAddrBuf[chip::Transport::PeerAddress::kMaxToStringSize];
+
+    VerifyOrReturn(conn != nullptr);
+
+    appConnCbCtxt = conn->mAppState;
+    VerifyOrReturn(appConnCbCtxt != nullptr);
+    Transport::PeerAddress peerAddr = conn->mPeerAddr;
+    peerAddr.ToString(peerAddrBuf);
+
+    if (appConnCbCtxt->connCompleteCb != nullptr)
+    {
+        appConnCbCtxt->connCompleteCb(conn, conErr);
+    }
+    else
+    {
+        ChipLogProgress(Inet, "%s: Connection established with peer %s, but no registered handler. Disconnecting.", peerAddrBuf,
+                        __func__);
+
+        // Close the connection
+        Disconnect(conn);
+    }
+}
+
+void SessionManager::HandleConnectionClosed(Transport::ActiveConnectionState * conn, CHIP_ERROR conErr)
+{
+    Transport::AppConnCallbackCtxt * appConnCbCtxt = nullptr;
+    VerifyOrReturn(conn != nullptr);
+
+    appConnCbCtxt = conn->mAppState;
+    VerifyOrReturn(appConnCbCtxt != nullptr);
+
+    // Mark the corresponding secure sessions as defunct
+    mSecureSessions.ForEachSession([&](auto session) {
+        if (session->IsActiveSession() && session->GetTCPConnection() == conn)
+        {
+            SessionHandle handle(*session);
+            // Notify the SessionConnection delegate of the connection
+            // closure.
+            if (mConnDelegate != nullptr)
+            {
+                mConnDelegate->OnConnectionClosed(handle, conErr);
+            }
+
+            // Dis-associate the connection from session by setting it to a
+            // nullptr.
+            session->SetTCPConnection(nullptr);
+            // Mark session as defunct
+            session->MarkAsDefunct();
+        }
+
+        return Loop::Continue;
+    });
+
+    // TODO: A mechanism to mark an unauthenticated session as unusable when
+    // the underlying connection is broken. Issue #32323
+
+    if (appConnCbCtxt->connClosedCb != nullptr)
+    {
+        appConnCbCtxt->connClosedCb(conn, conErr);
+    }
+}
+
+CHIP_ERROR SessionManager::ConnectToPeer(const PeerAddress & peerAddress, Transport::AppConnCallbackCtxt * appState)
+{
+    if (mTransportMgr != nullptr)
+    {
+        char peerAddrBuf[chip::Transport::PeerAddress::kMaxToStringSize];
+        peerAddress.ToString(peerAddrBuf);
+        ChipLogProgress(Inet, "Connecting to peer at %s.", peerAddrBuf);
```

Comment:
Should probably mention TCP in there somewhere.

Reply:
Added TCP in the log. The addresstype for PeerAddress emits the transport protocol in the log as well.